internal/shardchain/local_cache/hack.go

```diff
@@ -0,0 +1,22 @@
+package local_cache
```

Comment:
why this file called "hack"?

Reply:
This is because the unsafe package is used to improve performance, which may cause incompatibility in future go version upgrades (although most likely not)